src/extension/js/add-polyratings.js

```diff
@@ -205,6 +189,34 @@ export default class PolyratingIntegrator {
     })
   }
 
+  checkCalPolyRatings(rawname, nameElems) {
+    const [lastName, firstName] = rawname.split(/[, ]/)
+      .filter(name => name.length > 2 && name.split('I').length !== name.length + 1) // detects II and M.
+
+    const proxyUrl = 'https://cors-anywhere.herokuapp.com/'
+    const herf = `https://www.calpolyratings.com/${firstName.toLowerCase()}-${lastName.toLowerCase()}/`
+
+    fetch(proxyUrl + herf).then(res => res.text()).then(data => {
+      // Generate random string for dom element to avoid conflict in parsing
+      const domID = (new Array(10)).fill(null).map(_ => String.fromCharCode((Math.ceil(Math.random() * 26) + 96))).join('')
```

Comment:
Creating a random ID and a new element is unnecessary here. Once you have the response from calpolyratings, the workflow should just be to grab the ratings/eval from the returned html using css selectors to find the specific elements you want. It should look something like how I did it for the other site:
```
const ratingElem = polyratingPage.find('.hidden-xs > span > .text-primary')
if (ratingElem[0]) {
  const rating = ratingElem.text()
  const profId = response.match(/profid=(\d+)/)[1]
  const evals = ratingElem.next().text().replace('uation', '')
  const href = `http://polyratings.com/eval.php?profid=${profId}`
  const numericalRating = parseFloat(rating, 10)
  const bgColor = this.calculateBackgroundColor(numericalRating)
  this.updateInstructorName(rawName, nameElems, bgColor, href, rating, evals)
} else {
  // not found
}
```